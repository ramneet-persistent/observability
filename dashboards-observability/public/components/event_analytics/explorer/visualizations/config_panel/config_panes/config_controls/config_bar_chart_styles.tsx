/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useMemo, useCallback, Fragment } from 'react';
import { EuiAccordion, EuiSpacer } from '@elastic/eui';
import { get } from 'lodash';
import { ButtonGroupItem } from './config_button_group';
import { IConfigPanelOptionSection } from '../../../../../../../../common/types/explorer';
import {
  SLIDER_MIN_VALUE,
  SLIDER_MAX_VALUE,
  SLIDER_DEFAULT_STEP,
} from '../../../../../../../../common/constants/shared';
import { ConfigChartOptionsEnum } from '../../../../../../../../common/constants/explorer';

export const ConfigBarChartStyles = ({
  visualizations,
  schemas,
  vizState,
  handleConfigChange,
  sectionName,
  sectionId = 'chartStyles',
}: any) => {
  const { data } = visualizations;
  const { data: vizData = {}, metadata: { fields = [] } = {} } = data?.rawVizData;

  const handleConfigurationChange = useCallback(
    (stateFieldName) => {
      return (changes) => {
        handleConfigChange({
          ...vizState,
          [stateFieldName]: changes,
        });
      };
    },
    [handleConfigChange, vizState]
  );

  /* To update the schema options based on current style mode selection */
  const currentSchemas = useMemo(() => {
    if (vizState?.orientation === 'h') {
      return schemas.filter((schema: IConfigPanelOptionSection) => schema.mapTo !== 'labelAngle');
    }
    return schemas;
  }, [vizState]);

  const dimensions = useMemo(
    () =>
      currentSchemas.map((schema: IConfigPanelOptionSection, index: number) => {
        let params = {
          title: schema.name,
          vizState,
          ...schema.props,
        };
        const DimensionComponent = schema.component || ButtonGroupItem;

        const createDimensionComponent = (dimProps) => (
          <Fragment key={`viz-series-${index}`}>
            <DimensionComponent {...dimProps} />
            <EuiSpacer size="s" />
          </Fragment>
        );
        switch (schema.eleType) {
          case ConfigChartOptionsEnum.buttons:
            params = {
              ...params,
              legend: schema.name,
              groupOptions: schema?.props?.options.map((btn: { name: string }) => ({
                ...btn,
                label: btn.name,
              })),
              idSelected: vizState[schema.mapTo] || schema?.props?.defaultSelections[0]?.id,
              handleButtonChange: handleConfigurationChange(schema.mapTo),
            };
            return createDimensionComponent(params);
          case ConfigChartOptionsEnum.input:
            params = {
              title: schema.name,
              currentValue: vizState[schema.mapTo] || '',
              numValue: vizState[schema.mapTo] || '',
              handleInputChange: handleConfigurationChange(schema.mapTo),
              vizState,
              ...schema.props,
            };
            return createDimensionComponent(params);

          case ConfigChartOptionsEnum.slider:
            params = {
              ...params,
              minRange:
                typeof get(schema, 'props.min') === undefined
                  ? SLIDER_MIN_VALUE
                  : get(schema, 'props.min'),
              maxRange:
                typeof get(schema, 'props.max') === undefined
                  ? SLIDER_MAX_VALUE
                  : get(schema, 'props.max'),
              step: schema?.props?.step || SLIDER_DEFAULT_STEP,
              currentRange: vizState[schema.mapTo] || schema?.defaultState,
              ticks: schema?.props?.ticks,
              showTicks: schema?.props?.showTicks || false,
              handleSliderChange: handleConfigurationChange(schema.mapTo),
            };
            return createDimensionComponent(params);

          default:
            params = {
              ...params,
              paddingTitle: schema.name,
              advancedTitle: 'advancedTitle',
              dropdownList:
                schema?.options?.map((option) => ({ ...option })) ||
                fields.map((item) => ({ ...item })),
              onSelectChange: handleConfigurationChange(schema.mapTo),
              isSingleSelection: schema.isSingleSelection,
              selectedAxis: vizState[schema.mapTo] || schema.defaultState,
            };
            return createDimensionComponent(params);
        }
      }),
    [currentSchemas, vizState, handleConfigurationChange]
  );

  return (
    <EuiAccordion
      initialIsOpen
      id={`configPanel__${sectionId}`}
      buttonContent={sectionName}
      paddingSize="s"
    >
      {dimensions}
    </EuiAccordion>
  );
};
