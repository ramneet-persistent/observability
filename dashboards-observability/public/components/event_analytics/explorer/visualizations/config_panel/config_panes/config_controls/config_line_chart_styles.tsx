/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useMemo, useCallback, Fragment } from 'react';
import { EuiAccordion, EuiSpacer } from '@elastic/eui';
import { get } from 'lodash';
import { ButtonGroupItem } from './config_button_group';
import { IConfigPanelOptionSection } from '../../../../../../../../common/types/explorer';
import {
  visChartTypes,
  SLIDER_MIN_VALUE,
  SLIDER_MAX_VALUE,
  SLIDER_DEFAULT_STEP,
} from '../../../../../../../../common/constants/shared';
import { ConfigChartOptionsEnum } from '../../../../../../../../common/constants/explorer';

export const ConfigLineChartStyles = ({
  visualizations,
  schemas,
  vizState,
  handleConfigChange,
  sectionName,
  sectionId = 'chartStyles',
}: any) => {
  const { data } = visualizations;
  const { data: vizData = {}, metadata: { fields = [] } = {} } = data?.rawVizData;

  const handleConfigurationChange = useCallback(
    (stateFiledName) => {
      return (changes) => {
        handleConfigChange({
          ...vizState,
          [stateFiledName]: changes,
        });
      };
    },
    [handleConfigChange, vizState]
  );

  /* To update the schema options based on current style mode selection */
  const currentSchemas = useMemo(() => {
    if (vizState?.style) {
      switch (vizState.style) {
        case 'lines':
          return schemas.filter(
            (schema: IConfigPanelOptionSection) => schema.mapTo !== 'markerSize'
          );
        case 'bar':
          return schemas.filter(
            (schema: IConfigPanelOptionSection) =>
              !['interpolation', 'markerSize'].includes(schema.mapTo)
          );
        case 'markers':
          return schemas.filter((schema: IConfigPanelOptionSection) =>
            ['style', 'markerSize'].includes(schema.mapTo)
          );
        case 'lines+markers':
          return schemas.filter(
            (schema: IConfigPanelOptionSection) => schema.mapTo !== 'interpolation'
          );
      }
    } else if (visualizations?.vis?.name === visChartTypes.Scatter) {
      return schemas.filter((schema: IConfigPanelOptionSection) =>
        ['style', 'markerSize'].includes(schema.mapTo)
      );
    } else {
      return schemas.filter((schema: IConfigPanelOptionSection) => schema.mapTo !== 'markerSize');
    }
  }, [vizState]);

  const dimensions = useMemo(
    () =>
      currentSchemas &&
      currentSchemas.map((schema: IConfigPanelOptionSection, index: number) => {
        const DimensionComponent = schema.component || ButtonGroupItem;
        let params = {
          title: schema.name,
          vizState,
          ...schema.props,
        };

        switch (schema.eleType) {
          case ConfigChartOptionsEnum.buttons:
            params = {
              ...params,
              legend: schema.name,
              groupOptions: schema?.props?.options.map((btn: { name: string }) => ({
                ...btn,
                label: btn.name,
              })),
              idSelected: vizState[schema.mapTo] || schema?.props?.defaultSelections[0]?.id,
              handleButtonChange: handleConfigurationChange(schema.mapTo),
            };
            break;
          case ConfigChartOptionsEnum.slider:
            params = {
              ...params,
              minRange:
                typeof get(schema, 'props.min') === undefined
                  ? SLIDER_MIN_VALUE
                  : get(schema, 'props.min'),
              maxRange:
                typeof get(schema, 'props.max') === undefined
                  ? SLIDER_MAX_VALUE
                  : get(schema, 'props.max'),
              step: schema?.props?.step || SLIDER_DEFAULT_STEP,
              currentRange: vizState[schema.mapTo] || schema?.defaultState,
              ticks: schema?.props?.ticks,
              showTicks: schema?.props?.showTicks || false,
              handleSliderChange: handleConfigurationChange(schema.mapTo),
            };
            break;
          case ConfigChartOptionsEnum.input:
            params = {
              ...params,
              numValue: vizState[schema.mapTo] || '',
              handleInputChange: handleConfigurationChange(schema.mapTo),
            };
            break;

          default:
            params = {
              ...params,
              paddingTitle: schema.name,
              advancedTitle: 'advancedTitle',
              dropdownList:
                schema?.options?.map((option) => ({ ...option })) ||
                fields.map((item) => ({ ...item })),
              onSelectChange: handleConfigurationChange(schema.mapTo),
              isSingleSelection: schema.isSingleSelection,
              selectedAxis: vizState[schema.mapTo] || schema.defaultState,
            };
        }

        return (
          <Fragment key={`viz-series-${index}`}>
            <DimensionComponent {...params} />
            <EuiSpacer size="s" />
          </Fragment>
        );
      }),
    [currentSchemas, vizState, handleConfigurationChange]
  );

  return (
    <EuiAccordion
      initialIsOpen
      id={`configPanel__${sectionId}`}
      buttonContent={sectionName}
      paddingSize="s"
    >
      {dimensions}
    </EuiAccordion>
  );
};
