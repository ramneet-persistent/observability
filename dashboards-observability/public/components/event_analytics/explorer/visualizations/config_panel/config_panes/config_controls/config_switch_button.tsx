/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiSwitch, EuiFormRow, EuiSpacer } from '@elastic/eui';

interface SwitchButtonProps {
  currentValue: boolean;
  title: string;
  onToggle: (value: boolean) => void;
}
export const SwitchButton = ({
  currentValue,
  onToggle,
  title,
}: SwitchButtonProps) => {
  return (
    <>
      {/* <EuiFormRow display="columnCompressedSwitch"  label={title}> */}
      <EuiSpacer />
      <EuiSwitch
        label={title}
        checked={currentValue}
        onChange={(e) => onToggle(e.target.checked)}
      />
      {/* </EuiFormRow> */}
    </>
  );
};
