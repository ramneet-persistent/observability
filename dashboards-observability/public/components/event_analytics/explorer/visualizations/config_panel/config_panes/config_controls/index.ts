/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export { ConfigPanelOptions } from './config_panel_options';
export { ConfigValueOptions } from './config_value_options';
export { ConfigChartOptions } from './config_chart_options';
export { ConfigDataLinks } from './config_data_links';
export { ConfigThresholds } from './config_thresholds';
export { ConfigText } from './config_text';
export { ConfigGaugeValueOptions } from './config_gauge_options';
export { ConfigInputField } from "./config_input_field";
export { ConfigGridColumns } from "./config_grid_columns"
export { ConfigColorTheme } from './config_color_theme';

