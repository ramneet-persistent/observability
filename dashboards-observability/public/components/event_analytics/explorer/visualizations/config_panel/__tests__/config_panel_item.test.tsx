/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { PanelItem } from '../config_panes/config_controls/config_panel_item';

describe('Config panel item component', () => {
  configure({ adapter: new Adapter() });

  it('Renders PanelItem with empty data', async () => {
    const wrapper = mount(
      <PanelItem
        paddingTitle={'Test Title'}
        selectedAxis={[]}
        dropdownList={[]}
        onSelectChange={jest.fn()}
        isInvalid={true}
        isSingleSelection={false}
        isClearable={true}
      />
    );

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('Renders PanelItem with data', async () => {
    const wrapper = mount(
      <PanelItem
        paddingTitle={'Test Mode'}
        selectedAxis={[
          {
            name: 'spectrum',
            label: 'spectrum',
            value: 'spectrum',
          },
        ]}
        dropdownList={[
          {
            name: 'spectrum',
            label: 'spectrum',
            value: 'spectrum',
          },
          {
            name: 'opacity',
            label: 'opacity',
            value: 'opacity',
          },
        ]}
        onSelectChange={jest.fn()}
        isInvalid={false}
        isSingleSelection={false}
        isClearable={true}
      />
    );

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
