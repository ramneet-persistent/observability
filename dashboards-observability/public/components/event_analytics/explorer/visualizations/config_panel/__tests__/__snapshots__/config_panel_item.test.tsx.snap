// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config panel item component Renders PanelItem with data 1`] = `
<PanelItem
  dropdownList={
    Array [
      Object {
        "label": "spectrum",
        "name": "spectrum",
        "value": "spectrum",
      },
      Object {
        "label": "opacity",
        "name": "opacity",
        "value": "opacity",
      },
    ]
  }
  isClearable={true}
  isInvalid={false}
  isSingleSelection={false}
  onSelectChange={[MockFunction]}
  paddingTitle="Test Mode"
  selectedAxis={
    Array [
      Object {
        "label": "spectrum",
        "name": "spectrum",
        "value": "spectrum",
      },
    ]
  }
>
  <EuiTitle
    size="xxs"
  >
    <h3
      className="euiTitle euiTitle--xxsmall"
    >
      Test Mode
    </h3>
  </EuiTitle>
  <EuiSpacer
    size="s"
  >
    <div
      className="euiSpacer euiSpacer--s"
    />
  </EuiSpacer>
  <EuiComboBox
    aria-label="Use aria labels when no actual label is in use"
    async={false}
    compressed={false}
    fullWidth={false}
    id="axis-select-2"
    isClearable={true}
    isInvalid={false}
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "label": "spectrum",
          "name": "spectrum",
          "value": "spectrum",
        },
        Object {
          "label": "opacity",
          "name": "opacity",
          "value": "opacity",
        },
      ]
    }
    placeholder="Select a field"
    selectedOptions={
      Array [
        Object {
          "label": "spectrum",
          "name": "spectrum",
          "value": "spectrum",
        },
      ]
    }
    singleSelection={false}
    sortMatchesBy="none"
  >
    <div
      aria-expanded={false}
      aria-haspopup="listbox"
      aria-label="Use aria labels when no actual label is in use"
      className="euiComboBox"
      onKeyDown={[Function]}
      role="combobox"
    >
      <EuiComboBoxInput
        autoSizeInputRef={[Function]}
        compressed={false}
        fullWidth={false}
        hasSelectedOptions={true}
        id="axis-select-2"
        inputRef={[Function]}
        isListOpen={false}
        noIcon={false}
        onChange={[Function]}
        onClear={[Function]}
        onClick={[Function]}
        onCloseListClick={[Function]}
        onFocus={[Function]}
        onOpenListClick={[Function]}
        onRemoveOption={[Function]}
        placeholder="Select a field"
        rootId={[Function]}
        searchValue=""
        selectedOptions={
          Array [
            Object {
              "label": "spectrum",
              "name": "spectrum",
              "value": "spectrum",
            },
          ]
        }
        singleSelection={false}
        toggleButtonRef={[Function]}
        updatePosition={[Function]}
        value="spectrum"
      >
        <EuiFormControlLayout
          clear={
            Object {
              "data-test-subj": "comboBoxClearButton",
              "onClick": [Function],
            }
          }
          compressed={false}
          fullWidth={false}
          icon={
            Object {
              "aria-label": "Open list of options",
              "data-test-subj": "comboBoxToggleListButton",
              "disabled": undefined,
              "onClick": [Function],
              "ref": [Function],
              "side": "right",
              "type": "arrowDown",
            }
          }
        >
          <div
            className="euiFormControlLayout"
          >
            <div
              className="euiFormControlLayout__childrenWrapper"
            >
              <div
                className="euiComboBox__inputWrap euiComboBox__inputWrap-isClearable"
                data-test-subj="comboBoxInput"
                onClick={[Function]}
                tabIndex={-1}
              >
                <EuiComboBoxPill
                  asPlainText={false}
                  color="hollow"
                  key="spectrum"
                  name="spectrum"
                  onClose={[Function]}
                  option={
                    Object {
                      "label": "spectrum",
                      "name": "spectrum",
                      "value": "spectrum",
                    }
                  }
                  value="spectrum"
                >
                  <EuiI18n
                    default="Remove {children} from selection in this group"
                    token="euiComboBoxPill.removeSelection"
                    values={
                      Object {
                        "children": "spectrum",
                      }
                    }
                  >
                    <EuiBadge
                      className="euiComboBoxPill"
                      closeButtonProps={
                        Object {
                          "tabIndex": -1,
                        }
                      }
                      color="hollow"
                      iconOnClick={[Function]}
                      iconOnClickAriaLabel="Remove spectrum from selection in this group"
                      iconSide="right"
                      iconType="cross"
                      name="spectrum"
                      title="spectrum"
                      value="spectrum"
                    >
                      <EuiInnerText>
                        <span
                          className="euiBadge euiBadge--hollow euiBadge--iconRight euiComboBoxPill"
                          name="spectrum"
                          title="spectrum"
                          value="spectrum"
                        >
                          <span
                            className="euiBadge__content"
                          >
                            <span
                              className="euiBadge__text"
                            >
                              spectrum
                            </span>
                            <button
                              aria-label="Remove spectrum from selection in this group"
                              className="euiBadge__iconButton"
                              disabled={false}
                              onClick={[Function]}
                              title="Remove spectrum from selection in this group"
                              type="button"
                            >
                              <EuiIcon
                                className="euiBadge__icon"
                                color="inherit"
                                size="s"
                                tabIndex={-1}
                                type="cross"
                              >
                                <EuiIconEmpty
                                  aria-hidden={true}
                                  className="euiIcon euiIcon--small euiIcon--inherit euiIcon-isLoading euiBadge__icon"
                                  focusable="false"
                                  role="img"
                                  style={null}
                                  tabIndex={-1}
                                >
                                  <svg
                                    aria-hidden={true}
                                    className="euiIcon euiIcon--small euiIcon--inherit euiIcon-isLoading euiBadge__icon"
                                    focusable="false"
                                    height={16}
                                    role="img"
                                    style={null}
                                    tabIndex={-1}
                                    viewBox="0 0 16 16"
                                    width={16}
                                    xmlns="http://www.w3.org/2000/svg"
                                  />
                                </EuiIconEmpty>
                              </EuiIcon>
                            </button>
                          </span>
                        </span>
                      </EuiInnerText>
                    </EuiBadge>
                  </EuiI18n>
                </EuiComboBoxPill>
                <AutosizeInput
                  aria-controls=""
                  className="euiComboBox__input"
                  data-test-subj="comboBoxSearchInput"
                  id="axis-select-2"
                  injectStyles={true}
                  inputRef={[Function]}
                  minWidth={1}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="textbox"
                  style={
                    Object {
                      "fontSize": 14,
                    }
                  }
                  value=""
                >
                  <div
                    className="euiComboBox__input"
                    style={
                      Object {
                        "display": "inline-block",
                        "fontSize": 14,
                      }
                    }
                  >
                    <input
                      aria-controls=""
                      data-test-subj="comboBoxSearchInput"
                      id="axis-select-2"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      role="textbox"
                      style={
                        Object {
                          "boxSizing": "content-box",
                          "width": "2px",
                        }
                      }
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    />
                  </div>
                </AutosizeInput>
              </div>
              <EuiFormControlLayoutIcons
                clear={
                  Object {
                    "data-test-subj": "comboBoxClearButton",
                    "onClick": [Function],
                  }
                }
                compressed={false}
                icon={
                  Object {
                    "aria-label": "Open list of options",
                    "data-test-subj": "comboBoxToggleListButton",
                    "disabled": undefined,
                    "onClick": [Function],
                    "ref": [Function],
                    "side": "right",
                    "type": "arrowDown",
                  }
                }
              >
                <div
                  className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                >
                  <EuiFormControlLayoutClearButton
                    data-test-subj="comboBoxClearButton"
                    onClick={[Function]}
                    size="m"
                  >
                    <EuiI18n
                      default="Clear input"
                      token="euiFormControlLayoutClearButton.label"
                    >
                      <button
                        aria-label="Clear input"
                        className="euiFormControlLayoutClearButton"
                        data-test-subj="comboBoxClearButton"
                        onClick={[Function]}
                        type="button"
                      >
                        <EuiIcon
                          className="euiFormControlLayoutClearButton__icon"
                          type="cross"
                        >
                          <EuiIconEmpty
                            aria-hidden={true}
                            className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutClearButton__icon"
                            focusable="false"
                            role="img"
                            style={null}
                          >
                            <svg
                              aria-hidden={true}
                              className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutClearButton__icon"
                              focusable="false"
                              height={16}
                              role="img"
                              style={null}
                              viewBox="0 0 16 16"
                              width={16}
                              xmlns="http://www.w3.org/2000/svg"
                            />
                          </EuiIconEmpty>
                        </EuiIcon>
                      </button>
                    </EuiI18n>
                  </EuiFormControlLayoutClearButton>
                  <EuiFormControlLayoutCustomIcon
                    aria-label="Open list of options"
                    data-test-subj="comboBoxToggleListButton"
                    iconRef={[Function]}
                    onClick={[Function]}
                    size="m"
                    type="arrowDown"
                  >
                    <button
                      aria-label="Open list of options"
                      className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                      data-test-subj="comboBoxToggleListButton"
                      onClick={[Function]}
                      type="button"
                    >
                      <EuiIcon
                        aria-hidden="true"
                        className="euiFormControlLayoutCustomIcon__icon"
                        size="m"
                        type="arrowDown"
                      >
                        <EuiIconArrowDown
                          aria-hidden={true}
                          className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                          focusable="false"
                          role="img"
                          style={null}
                        >
                          <svg
                            aria-hidden={true}
                            className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                            focusable="false"
                            height={16}
                            role="img"
                            style={null}
                            viewBox="0 0 16 16"
                            width={16}
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M13.069 5.157L8.384 9.768a.546.546 0 01-.768 0L2.93 5.158a.552.552 0 00-.771 0 .53.53 0 000 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 000-.76.552.552 0 00-.771 0z"
                              fillRule="non-zero"
                            />
                          </svg>
                        </EuiIconArrowDown>
                      </EuiIcon>
                    </button>
                  </EuiFormControlLayoutCustomIcon>
                </div>
              </EuiFormControlLayoutIcons>
            </div>
          </div>
        </EuiFormControlLayout>
      </EuiComboBoxInput>
    </div>
  </EuiComboBox>
</PanelItem>
`;

exports[`Config panel item component Renders PanelItem with empty data 1`] = `
<PanelItem
  dropdownList={Array []}
  isClearable={true}
  isInvalid={true}
  isSingleSelection={false}
  onSelectChange={[MockFunction]}
  paddingTitle="Test Title"
  selectedAxis={Array []}
>
  <EuiTitle
    size="xxs"
  >
    <h3
      className="euiTitle euiTitle--xxsmall"
    >
      Test Title
    </h3>
  </EuiTitle>
  <EuiSpacer
    size="s"
  >
    <div
      className="euiSpacer euiSpacer--s"
    />
  </EuiSpacer>
  <EuiComboBox
    aria-label="Use aria labels when no actual label is in use"
    async={false}
    compressed={false}
    fullWidth={false}
    id="axis-select-1"
    isClearable={true}
    isInvalid={true}
    onChange={[MockFunction]}
    options={Array []}
    placeholder="Select a field"
    selectedOptions={Array []}
    singleSelection={false}
    sortMatchesBy="none"
  >
    <div
      aria-expanded={false}
      aria-haspopup="listbox"
      aria-label="Use aria labels when no actual label is in use"
      className="euiComboBox euiComboBox-isInvalid"
      onKeyDown={[Function]}
      role="combobox"
    >
      <EuiComboBoxInput
        autoSizeInputRef={[Function]}
        compressed={false}
        fullWidth={false}
        hasSelectedOptions={false}
        id="axis-select-1"
        inputRef={[Function]}
        isListOpen={false}
        noIcon={false}
        onChange={[Function]}
        onClear={[Function]}
        onClick={[Function]}
        onCloseListClick={[Function]}
        onFocus={[Function]}
        onOpenListClick={[Function]}
        onRemoveOption={[Function]}
        placeholder="Select a field"
        rootId={[Function]}
        searchValue=""
        selectedOptions={Array []}
        singleSelection={false}
        toggleButtonRef={[Function]}
        updatePosition={[Function]}
        value=""
      >
        <EuiFormControlLayout
          compressed={false}
          fullWidth={false}
          icon={
            Object {
              "aria-label": "Open list of options",
              "data-test-subj": "comboBoxToggleListButton",
              "disabled": undefined,
              "onClick": [Function],
              "ref": [Function],
              "side": "right",
              "type": "arrowDown",
            }
          }
        >
          <div
            className="euiFormControlLayout"
          >
            <div
              className="euiFormControlLayout__childrenWrapper"
            >
              <div
                className="euiComboBox__inputWrap euiComboBox__inputWrap-isClearable"
                data-test-subj="comboBoxInput"
                onClick={[Function]}
                tabIndex={-1}
              >
                <p
                  className="euiComboBoxPlaceholder"
                >
                  Select a field
                </p>
                <AutosizeInput
                  aria-controls=""
                  className="euiComboBox__input"
                  data-test-subj="comboBoxSearchInput"
                  id="axis-select-1"
                  injectStyles={true}
                  inputRef={[Function]}
                  minWidth={1}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="textbox"
                  style={
                    Object {
                      "fontSize": 14,
                    }
                  }
                  value=""
                >
                  <div
                    className="euiComboBox__input"
                    style={
                      Object {
                        "display": "inline-block",
                        "fontSize": 14,
                      }
                    }
                  >
                    <input
                      aria-controls=""
                      data-test-subj="comboBoxSearchInput"
                      id="axis-select-1"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      role="textbox"
                      style={
                        Object {
                          "boxSizing": "content-box",
                          "width": "2px",
                        }
                      }
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    />
                  </div>
                </AutosizeInput>
              </div>
              <EuiFormControlLayoutIcons
                compressed={false}
                icon={
                  Object {
                    "aria-label": "Open list of options",
                    "data-test-subj": "comboBoxToggleListButton",
                    "disabled": undefined,
                    "onClick": [Function],
                    "ref": [Function],
                    "side": "right",
                    "type": "arrowDown",
                  }
                }
              >
                <div
                  className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                >
                  <EuiFormControlLayoutCustomIcon
                    aria-label="Open list of options"
                    data-test-subj="comboBoxToggleListButton"
                    iconRef={[Function]}
                    onClick={[Function]}
                    size="m"
                    type="arrowDown"
                  >
                    <button
                      aria-label="Open list of options"
                      className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                      data-test-subj="comboBoxToggleListButton"
                      onClick={[Function]}
                      type="button"
                    >
                      <EuiIcon
                        aria-hidden="true"
                        className="euiFormControlLayoutCustomIcon__icon"
                        size="m"
                        type="arrowDown"
                      >
                        <EuiIconEmpty
                          aria-hidden={true}
                          className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                          focusable="false"
                          role="img"
                          style={null}
                        >
                          <svg
                            aria-hidden={true}
                            className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                            focusable="false"
                            height={16}
                            role="img"
                            style={null}
                            viewBox="0 0 16 16"
                            width={16}
                            xmlns="http://www.w3.org/2000/svg"
                          />
                        </EuiIconEmpty>
                      </EuiIcon>
                    </button>
                  </EuiFormControlLayoutCustomIcon>
                </div>
              </EuiFormControlLayoutIcons>
            </div>
          </div>
        </EuiFormControlLayout>
      </EuiComboBoxInput>
    </div>
  </EuiComboBox>
</PanelItem>
`;
